scripts/core/spellcheck/spellcheck.js

```diff
@@ -431,56 +435,60 @@ function SpellcheckService($q, api, dictionaries, $rootScope, $location, _) {
         return !isNaN(word) || !dict.content || isIgnored(word) || !isSpellingMistake(word, isSentenceWord);
     };
 
-
     // reset ignore list for an item if it was unlocked
     $rootScope.$on('item:unlock', (event, data) => {
         if (ignored.hasOwnProperty(data.item)) {
             ignored[data.item] = {};
         }
     });
-}
 
-SpellcheckMenuController.$inject = ['$rootScope', 'editorResolver', 'preferencesService'];
-function SpellcheckMenuController($rootScope, editorResolver, preferencesService) {
-    this.isAuto = null;
-    this.spellcheck = spellcheck;
-    this.pushSettings = pushSettings;
-    var PREFERENCES_KEY = 'spellchecker:status';
-    var self = this;
+    /**
+     * Get the spell checker status
+     */
+    this.getSpellcheckerStatus = function getSpellcheckerStatus() {
+        var status = true;
+
+        return preferencesService.get(PREFERENCES_KEY).then((result) => {
+            if (angular.isDefined(result)) {
+                status = result.enabled;
+            }
+            return status;
+        }, (error) => status);
+    };
 
     /**
      * Set the spell checker status
      */
-    function setStatus(status) {
+    this.setSpellcheckerStatus = function setSpellcheckerStatus(status) {
         var updates = {};
 
+        self.isAutoSpellchecker = status;
+
         updates[PREFERENCES_KEY] = {
             type: 'bool',
             enabled: status,
             default: true
         };
 
         preferencesService.update(updates, PREFERENCES_KEY);
-    }
+    };
 
-    /**
-     * Get the spell checker status
-     */
-    function getStatus() {
-        var status = true;
+    this.getSpellcheckerStatus().then((status) => {
+        self.isAutoSpellchecker = status;
+    });
+}
 
-        return preferencesService.get(PREFERENCES_KEY).then((result) => {
-            if (angular.isDefined(result)) {
-                status = result.enabled;
-            }
-            return status;
-        }, (error) => status);
-    }
+SpellcheckMenuController.$inject = ['$rootScope', 'editorResolver', 'spellcheck'];
+function SpellcheckMenuController($rootScope, editorResolver, spellcheck) {
+    this.isAuto = false;
+    this.runSpellchecker = runSpellchecker;
+    this.pushSettings = pushSettings;
+    var self = this;
 
     /**
      * Force spell ckecking
      */
-    function spellcheck() {
+    function runSpellchecker() {
```

Comment:
why is the name changed here?

Reply:
I use also spellcheck service and I wanted to avoid any confusion or name clash.